### DISCUSSION TITLE
Should i commit custom store?

### DISCUSSION THREAD

**Discussion by vincent-thomas** [pnpm/pnpm#4620] [Q&A]
Hey! I have this .npmrc:
```npmrc
auto-install-peers=true
store-dir="./cache/pnpm"
```
Should i commit the folder `./cache/pnpm`?
It really speeds installing speeds up and i can really see the benefits in CI/CD.
Are there any best practises?

**Comment by zkochan** [comment]
It is not a good idea because on a dev machine you have one store per disk, so all the projects use the same store and node_modules consume less disk space. This way, if you make the store part of your project, you don't get the benefits of less disk space usage.

**Reply by vincent-thomas** [reply]
> It is not a good idea because on a dev machine you have one store per disk, so all the projects use the same store and node_modules consume less disk space. This way, if you make the store part of your project, you don't get the benefits of less disk space usage.

Sorry i forgot to put this information in the discussion. I run a monorepo with all my serious apps and libraries. This way it doesn't really matter with the disk-storage issue and it takes a really long time to do ci/cd with all my projects with `npm install`. It takes like 20 seconds with this custom store method.

With this in mind, is it still a bad idea to commit `custom-store` @zkochan ?

**Reply by phawxby** [reply]
We took an alternative view of it when viewing yarn that we didn't so much care about disk space but we cared about build stability so we went with this model
https://medium.com/slackernoon/speed-up-ci-builds-with-npm-offline-cache-aa72a2bb63c5

I think committing the store to the repo is a reasonable use-case.
